src/qml/QFieldSettings.qml

```diff
@@ -1248,21 +1225,79 @@ Page {
                   }
 
                   ComboBox {
+                      id: verticalGridShiftComboBox
                       Layout.fillWidth: true
                       Layout.columnSpan: 2
-                      model: [ qsTr( "None" ) ].concat( platformUtilities.availableGrids() );
                       font: Theme.defaultFont
 
                       popup.font: Theme.defaultFont
                       popup.topMargin: mainWindow.sceneTopMargin
                       popup.bottomMargin: mainWindow.sceneTopMargin
 
-                      onCurrentIndexChanged: {
-                          positioningSettings.verticalGrid = currentIndex > 0 ? platformUtilities.availableGrids()[currentIndex - 1] : '';
+                      textRole: "text"
+                      valueRole: "value"
+
+                      model: ListModel {
+                          id: model
                       }
 
-                      Component.onCompleted: {
-                          currentIndex = positioningSettings.verticalGrid != '' ? find(positioningSettings.verticalGrid) : 0;
+                      onCurrentValueChanged: {
+                          if(!currentValue)
+                              return
+
+                          positioningSettings.elevationCorrectionMode = currentValue;
+
+                          if(positioningSettings.elevationCorrectionMode === Positioning.ElevationCorrectionMode.FromGeoidFile)
+                              positioningSettings.verticalGrid = currentText
+                          else
+                              positioningSettings.verticalGrid = ""
+                      }
+
+                      Component.onCompleted: reload()
+
+                      function reload()
+                      {
+                          verticalGridShiftComboBox.model.clear()
+                          verticalGridShiftComboBox.model.append({text: qsTr( "None" ), value: Positioning.ElevationCorrectionMode.None});
+
+                          if(positionSource.device.capabilities() & AbstractGnssReceiver.OrthometricAltitude)
+                              verticalGridShiftComboBox.model.append({text: qsTr( "Orthometric from device" ), value: Positioning.ElevationCorrectionMode.OrthometricFromDevice});
+
+                          // Add geoid files to combobox
+                          var geoidFiles = platformUtilities.availableGrids()
+                          for(var i = 0; i < geoidFiles.length; i++)
+                              verticalGridShiftComboBox.model.append( { text: geoidFiles[i], value: Positioning.ElevationCorrectionMode.FromGeoidFile } );
+
+                          if(positioningSettings.elevationCorrectionMode === Positioning.ElevationCorrectionMode.None)
```

Comment:
Can you add a space in between {if,for,etc.} and the opening parenthesis through your changes?